### PR TITLE
[FR005]Boost Beast integration.

### PR DESCRIPTION
**Pull request by wiatrM**
Let's merge as boost beast can be completed. We will do listeners and REST endpoint (handler) as a separate feature.

Please review all the changes.

---
Todo:
- [x] Listener
- [x] Http connection
- [x] Server class
- [x] Connection timeout (read)
- [x] Connection timeout (write)

### CODE REVIEW

File: kazesl/test/server/impl/TestServer.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+        ASSERT_TRUE(server != nullptr);
+
+        std::vector<Address> serverPort(1);
+        serverPort.back().ip =
+            boost::asio::ip::address::from_string("127.0.0.1"),
+        serverPort.back().port = 0;
+
+        auto ep = server->endpoints(serverPort);
+        ASSERT_EQ(ep.size(), 1);
```
Comment (wiatrM) [inline kazesl/test/server/impl/TestServer.cpp]: Yeah, I know. More test will be written when REST endpoint and true listener will be ready.

**Comment by lapek** [comment]
Why a external/json/json.hpp is added to repository? Try tu add it once again as a submodule - link to repo is in that file' header.

**Comment by wiatrM** [comment]
Reworked json package as @lapek described. Added connection timeout feature (for now only read). Added WIP flag to complete write timeout on connection.

**Comment by wiatrM** [comment]
@lapek could we mege?

**Pull request merged**